include/deal.II/numerics/fe_field_convolution_function.h

```diff
@@ -0,0 +1,280 @@
+   *
+   * @ingroup functions
+   * @author Luca Heltai, 2019
+   */
+  template <int dim,
+            int spacedim            = dim,
+            typename DoFHandlerType = DoFHandler<dim, spacedim>,
+            typename VectorType     = Vector<double>>
+  class FEFieldConvolutionFunction
+    : public Function<spacedim, typename VectorType::value_type>
+  {
+  public:
+    /**
+     * Construct a vector function. A smart pointers is stored to the dof
+     * handler and to the cache object, so you have to make sure they remain
+     * valid for the entire lifetime of this object. The number of
+     * components of this functions is equal to the number of components of the
+     * finite element object.
+     */
+    FEFieldConvolutionFunction(const DoFHandlerType &                 dh,
+                               const GridTools::Cache<dim, spacedim> &cache,
+                               const VectorType &data_vector,
+                               Functions::CutOffFunctionBase<spacedim> &kernel,
+                               const Quadrature<dim> &quadrature);
+
+    /**
+     * Get one vector value at the given point. It is inefficient to use
+     * single points. If you need more than one at a time, use the
+     * vector_value_list() function.
+     */
+    virtual void
+    vector_value(
+      const Point<spacedim> &                  p,
+      Vector<typename VectorType::value_type> &values) const override;
+
+    /**
+     * Return the value of the function at the given point. Unless there is
+     * only one component (i.e. the function is scalar), you should state the
+     * component you want to have evaluated; it defaults to zero, i.e. the
+     * first component. It is inefficient to use single points. If you need
+     * more than one at a time, use the vector_value_list function.
+     */
+    virtual typename VectorType::value_type
+    value(const Point<spacedim> &p,
+          const unsigned int     component = 0) const override;
+
+    /**
+     * Set @p values to the point values of the specified component of the
+     * function at the @p points. It is assumed that @p values already has the
+     * right size, i.e. the same size as the points array.
+     */
+    virtual void
+    value_list(const std::vector<Point<spacedim>> &          points,
+               std::vector<typename VectorType::value_type> &values,
+               const unsigned int component = 0) const override;
+
+
+    /**
+     * Set @p values to the point values of the function at the @p points. It
+     * is assumed that @p values already has the right size, i.e. the same
+     * size as the points array. This is rather efficient if all the points
+     * lie on the same cell. If this is not the case, things may slow down a
+     * bit.
+     *
+     * @note When using this function on a
+     * parallel::distributed::Triangulation you may get an exception when
+     * trying to evaluate the solution at a point that lies on an artificial
```

Comment:
isn't this in contradiction to the text in the introduction?

Reply:
also, the same restriction would apply to the other `value_*` functions and should be repeated there

Reply:
yes. This is no longer true. The introduction contains the correct version of the text.